src/odemis/gui/cont/streams.py

```diff
@@ -3323,7 +3323,8 @@ def clear_feature_streams(self):
             self._stream_bar.remove_stream_panel(sc.stream_panel)
             if hasattr(v, "removeStream"):
                 v.removeStream(sc.stream)
-            self.stream_controllers.remove(sc)
+            else:
```

Comment:
I don't think that's relate to whether the view can remove the stream or not.
I'd suggest to just explicitly check for whether the element is in the list:
`if sc in self.stream_controllers:`